### PR TITLE
v1.2

### PR DESCRIPTION
**Pull request by kierajreed**
bigger and better

btw removed one commit sry bout that

### CODE REVIEW

File: modlist.txt [hunk 1]
```diff
@@ -1,6 +0,0 @@
-akashictome
-botania
-morphtool
-psi
-potionfingers
-quark
```
Comment (bl4ckscor3) [inline modlist.txt]: Please leave this file in for legacy support.

File: src/main/java/bl4ckscor3/mod/isamodbyvazkii/IsAModByVazkii.java [hunk 2]
```diff
@@ -10,35 +10,45 @@
...
+		Map<String, ModContainer> modList = Loader.instance().getIndexedModList();
 
 		for(String modid : modList.keySet())
 		{
 			String authorList = modList.get(modid).getMetadata().getAuthorList();
 
-			if(authorList.toLowerCase().contains("vazkii"))
+			if(authorList.toLowerCase().contains("vazkii")) {
 				MODS.put(modid, modList.get(modid).getName());
+			}
```
Comment (bl4ckscor3) [inline src/main/java/bl4ckscor3/mod/isamodbyvazkii/IsAModByVazkii.java]: Please remove these curly brackets.

File: src/main/java/bl4ckscor3/mod/isamodbyvazkii/IsAModByVazkii.java [hunk 3]
```diff
@@ -10,35 +10,45 @@
...
+
+			if(modList.get(modid) instanceof FMLModContainer) {
```
Comment (bl4ckscor3) [inline src/main/java/bl4ckscor3/mod/isamodbyvazkii/IsAModByVazkii.java]: Curly bracket on new line, same with the one below.

File: src/main/java/bl4ckscor3/mod/isamodbyvazkii/IsAModByVazkii.java [hunk 4]
```diff
@@ -80,7 +90,7 @@ public static void onItemTooltip(ItemTooltipEvent event)
 			{
 				if(TextFormatting.getTextWithoutFormattingCodes(tooltips.get(i)).equals(modName))
 				{
-					tooltips.set(i, I18n.format("isamodbyvazkii:mod", tooltips.get(i)));
+					tooltips.set(i, I18n.format(MOD, tooltips.get(i)));
```
Comment (bl4ckscor3) [inline src/main/java/bl4ckscor3/mod/isamodbyvazkii/IsAModByVazkii.java]: What happens when you remove these lines, does the blue mod name still render with "is a mod by Vazkii"? https://github.com/bl4ckscor3/IsAModByVazkii/blob/master/src/main/java/bl4ckscor3/mod/isamodbyvazkii/IsAModByVazkii.java#L79-L86

File: src/main/java/bl4ckscor3/mod/isamodbyvazkii/IsAModByVazkii.java [hunk 5]
```diff
@@ -10,35 +10,45 @@
...
 		{
 			String authorList = modList.get(modid).getMetadata().getAuthorList();
 
-			if(authorList.toLowerCase().contains("vazkii"))
+			if(authorList.toLowerCase().contains("vazkii")) {
 				MODS.put(modid, modList.get(modid).getName());
+			}
+
+			if(modList.get(modid) instanceof FMLModContainer) {
+				if(modList.get(modid).getMetadata().getAuthorList().toLowerCase().contains("vazkii")) {
```
Comment (bl4ckscor3) [inline src/main/java/bl4ckscor3/mod/isamodbyvazkii/IsAModByVazkii.java]: You have an authorList variable above, please use it.

File: src/main/java/bl4ckscor3/mod/isamodbyvazkii/IsAModByVazkii.java [hunk 6]
```diff
@@ -10,35 +10,45 @@
...
+					FMLModContainer container = (FMLModContainer) modList.get(modid);
+					ModMetadata metadata = ReflectionHelper.getPrivateValue(FMLModContainer.class, container, "modMetadata");
+					metadata.name = I18n.format(MOD, metadata.name);
```
Comment (bl4ckscor3) [inline src/main/java/bl4ckscor3/mod/isamodbyvazkii/IsAModByVazkii.java:50]: New line above.

**Review by bl4ckscor3 — CHANGES_REQUESTED** [review]

File: src/main/java/bl4ckscor3/mod/isamodbyvazkii/IsAModByVazkii.java (hunk 6 above)
Comment (kierajreed) [reply src/main/java/bl4ckscor3/mod/isamodbyvazkii/IsAModByVazkii.java:50]: Above the `if` or in the `if`? Or the curly brace?

File: src/main/java/bl4ckscor3/mod/isamodbyvazkii/IsAModByVazkii.java (hunk 4 above)
Comment (kierajreed) [reply src/main/java/bl4ckscor3/mod/isamodbyvazkii/IsAModByVazkii.java]: it changes the mod's actual name so it should work

File: src/main/java/bl4ckscor3/mod/isamodbyvazkii/IsAModByVazkii.java (hunk 6 above)
Comment (bl4ckscor3) [reply src/main/java/bl4ckscor3/mod/isamodbyvazkii/IsAModByVazkii.java:50]: Between `ModMetadata metadata = ...` and the next line.

File: src/main/java/bl4ckscor3/mod/isamodbyvazkii/IsAModByVazkii.java [hunk 7]
```diff
@@ -37,12 +37,13 @@ public void postInit(FMLPostInitializationEvent event)
...
+			if(authorList.toLowerCase().contains("vazkii")) 
 				MODS.put(modid, modList.get(modid).getName());
-			}
 
-			if(modList.get(modid) instanceof FMLModContainer) {
-				if(modList.get(modid).getMetadata().getAuthorList().toLowerCase().contains("vazkii")) {
+			if(modList.get(modid) instanceof FMLModContainer) 
+      {
+				if(authorList.toLowerCase().contains("vazkii")) 
+        {
```
Comment (bl4ckscor3) [inline src/main/java/bl4ckscor3/mod/isamodbyvazkii/IsAModByVazkii.java]: Could you indent these braces correctly, please?

File: src/main/java/bl4ckscor3/mod/isamodbyvazkii/IsAModByVazkii.java (hunk 4 above)
Comment (bl4ckscor3) [reply src/main/java/bl4ckscor3/mod/isamodbyvazkii/IsAModByVazkii.java]: Can you try it out and if it works actually remove the lines, please?

File: src/main/java/bl4ckscor3/mod/isamodbyvazkii/IsAModByVazkii.java (hunk 7 above)
Comment (kierajreed) [reply src/main/java/bl4ckscor3/mod/isamodbyvazkii/IsAModByVazkii.java]: right... Vim did an oof

File: src/main/java/bl4ckscor3/mod/isamodbyvazkii/IsAModByVazkii.java (hunk 4 above)
Comment (kierajreed) [reply src/main/java/bl4ckscor3/mod/isamodbyvazkii/IsAModByVazkii.java]: yes

**Comment by kierajreed** [comment]
everything works now

File: src/main/java/bl4ckscor3/mod/isamodbyvazkii/IsAModByVazkii.java [hunk 8]
```diff
@@ -37,16 +37,13 @@ public void postInit(FMLPostInitializationEvent event)
 		{
 			String authorList = modList.get(modid).getMetadata().getAuthorList();
 
-			if(authorList.toLowerCase().contains("vazkii")) 
-				MODS.put(modid, modList.get(modid).getName());
-
```
Comment (bl4ckscor3) [inline src/main/java/bl4ckscor3/mod/isamodbyvazkii/IsAModByVazkii.java]: Is this removal intended? I believe it won't work trying to get any keys/values if you don't populate the Map 😅

Comment (kierajreed) [reply src/main/java/bl4ckscor3/mod/isamodbyvazkii/IsAModByVazkii.java]: oh right, need it for items/blocks

sorry lol

**Comment by bl4ckscor3** [comment]
Looking good.

**Review by bl4ckscor3 — APPROVED** [review]
